docs/architecture/adr-013-msg-authorization.md

```diff
@@ -0,0 +1,220 @@
+# ADR 13: Msg Authorization
+
+## Changelog
+
+- 2019-11-06: Initial ADR
+
+## Context
+
+At Hackatom Berlin in June 2019, the Gaians team worked on a `delegation` module that is described in the spec below.
+Prior to that, the B-Harvest team had worked on a [subkeys spec](https://github.com/cosmos/cosmos-sdk/issues/4480) that
+covered similar use cases. In discussions after the Hackathon, the `delegation` module approach was deemed to
+be more generic and there was community desire for the work to continue. This resulted in an ICF grant to continue
+this work along with the fee grants and key groups modules.
+
+## Decision
+
+We will create a module named `msg_authorization`
+
+The `msg_authorization` module provides support for granting arbitrary capabilities
```

Comment:
We will create a module named `msg_authorization`. The `msg_authorization` module provides ...